core/src/main/scala/com/eharmony/spotz/backend/SparkFunctions.scala

```diff
@@ -0,0 +1,93 @@
+package com.eharmony.spotz.backend
+
+import com.eharmony.spotz.objective.Objective
+import com.eharmony.spotz.optimizer.grid.Grid
+import com.eharmony.spotz.optimizer.hyperparam.RandomSampler
+import org.apache.spark.SparkContext
+
+import scala.reflect.ClassTag
+
+/**
+  * Use Spark to paralellize the optimizer computation.
+  */
+trait SparkFunctions extends BackendFunctions {
+  @transient val sc: SparkContext
+
+  /**
+    * This function will sample batchSize trial points starting with the given
+    * trial start index.  Each trial point will be generated from a rng seeded
+    * by (trial number + seed).  This ensures the any randomness is repeatable
+    * for future trial runs of the same seed.  Sampling occurs on the Spark worker
+    * nodes.
+    *
+    * @param startIndex the starting trial index
+    * @param batchSize the number of points to sample beginning at the startIndex
+    * @param objective the function on which a point is applied
+    * @param reducer the reducer function to use on all the various points and losses
+    *                generated
+    * @param hyperParams defined hyper parameters from which to sample
+    * @param seed the seed
+    * @param sampleFunction sample a point from the defined hyper parameter space
+    * @tparam P point type
+    * @tparam L loss type
+    * @return the best point with the best loss as a tuple
+    */
+  protected override def bestRandomPointAndLoss[P, L](
+                                                       startIndex: Long,
+                                                       batchSize: Long,
+                                                       objective: Objective[P, L],
+                                                       reducer: ((P, L), (P, L)) => (P, L),
+                                                       hyperParams: Map[String, RandomSampler[_]],
+                                                       seed: Long = 0,
+                                                       sampleFunction: (Map[String, RandomSampler[_]], Long) => P): (P, L) = {
+
+    assert(batchSize > 0, "batchSize must be greater than 0")
```

Comment:
Consider `require` instead of `assert` as argued [here](http://stackoverflow.com/questions/26140757/what-to-choose-between-require-and-assert-in-scala).